pkg/quota/v1/install/update_filter.go

```diff
@@ -30,6 +34,10 @@ func DefaultUpdateFilter() func(resource schema.GroupVersionResource, oldObj, ne
 		case schema.GroupResource{Resource: "pods"}:
 			oldPod := oldObj.(*v1.Pod)
 			newPod := newObj.(*v1.Pod)
+			// when AllocatedResources changed
+			if feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && hasAllocatedResourcesChanged(oldPod, newPod) {
+				return true
+			}
```

Comment:
nit:
```suggestion
			if feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
			         // when AllocatedResources changed
			         return hasAllocatedResourcesChanged(oldPod, newPod) {
			}
```

Reply:
At your suggestion we will not execute the rest of the logic (next line): 
-> return core.QuotaV1Pod(oldPod, clock.RealClock{}) && !core.QuotaV1Pod(newPod, clock.RealClock{}) 
( when feature enabled and  NOT hasAllocatedResourcesChanged  -> need to check QuotaV1Pod oldPod and newPod )

Reply:
True. I take it back.